### ISSUE TITLE
SIGMA to ElastAlert bash script

### ISSUE THREAD

**Issue by Jackson-Pollock** [SigmaHQ/sigma#1815]
Hi,
I'm working on bash script to convert misp events to sigma rules using [sigmai](https://github.com/0xThiebaut/sigmai/tree/8e0171c00e1a4af1782f3992a61c5dc777624336) and then sigma to elastalert using sigmac, so the sigmai command will convert misp event to sigma rule and store as random filename `uiid.yml` (user can't define filename). So I want to use this filename as variable and call in sigmac command as input.

sigmai o/p:
terminal#`2021-08-09T09:54:31Z INF saved Sigma rule rule=60118dab-1ab8-40b2-b02b-b6f80aba047c`
This will save file called `60118dab-1ab8-40b2-b02b-b6f80aba047c.yml` in given path and now I want to use this file `60118dab-1ab8-40b2-b02b-b6f80aba047c.yml` as I input in sigmac command to convert from sigma to elastalert. 

Below is the snip code from shell script.

```
# if misp_to_sigma_rules enabled, then get MISP events to SIGMA rules
if [[ $misp_to_sigma_rules == "yes" ]];then
	#Convert MISP events to SIGMA using sigmai tool
	#git clone https://github.com/0xThiebaut/sigmai.git
	echo
	echo "converting MISP events to SIGMA rules"
	echo
	/root/sigmai/sigmai_linux_amd64 -t directory --directory-path /root/sigmai/rules/mispsigma -s misp --misp-url $mispurl --misp-key $apikey --misp-events $misp_event_id
else
	:
fi

#read output rule=value and create variable as $how_file_name

# if sigma_to_elastalert_rules enabled, then convert SIGMA to ElastAlert
if [[ $sigma_to_elastalert_rules == "yes" ]];then
	#git clone https://github.com/SigmaHQ/sigma.git
	echo
	echo "Converting SIGMA to ElastAlert"
	echo
    /root/sigma/tools/sigmac -t elastalert -c tools/config/winlogbeat-modules-enabled.yml --backend-option keyword_field="" -o /root/sigmai/rules/mispsigma/$event_id.yml /root/sigmai/rules/mispsigma/mispevent$how_file_name.yml
else
	:
fi
```
Please suggest how can I define `$how_file_name` here.

**Comment by Jackson-Pollock** [comment]
I know this is not a issue or not even fully in scope of sigmac but trying to get it solved.

**Comment by weslambert** [comment]
The following has worked for me to some degree, although it probably needs further refining:

```
#!/bin/bash
MISP="https://localhost"
APIKEY="ABCD1234"
SIGMAI="/home/user/sigmai"
SIGMAC="/home/user/sigma/tools/sigmac"
SIGMAPATH="/home/user/sigma"
SIGMAC="$SIGMAPATH/tools/sigmac"
SIGMACONFIGPATH="$SIGMAPATH/tools/config"
SIGMARULES="/home/user/sigmarules"
ELASTALERTRULES="/opt/so/rules/elastalert/misp"
CONFIGOPTIONS="-c /home/user/securityonion-baseline.yml"
BACKENDOPTIONS="--backend-option keyword_whitelist=source.ip,destination.ip,source.port,destination.port --backend-option keyword_field=.keyword --backend-option analyzed_sub_field_name=.security --backend-option wildcard_use_keyword=false"
MISP_INSECURE="--misp-insecure"
MISP_FILTER="--misp-levels 1,2"

# Pre-create these dirs
mkdir -p "$SIGMARULES"
mkdir -p "$ELASTALERTRULES"

# Create Sigma rules from MISP
echo "Creating Sigma rules from MISP events..."
"$SIGMAI" -t directory --directory-path "$SIGMARULES" -s misp --misp-url "$MISP" --misp-key "$APIKEY" $MISP_FILTER "$MISP_INSECURE"

# Loop through all Sigma rules, convert to Elastalert rules, and split the files
# Once support is merged into sigmac, csplit usage will be removed.
echo "Converting MISP Sigma rules to Elastalert rules..."
cd $ELASTALERTRULES
for i in $(ls "$SIGMARULES"); do
        "$SIGMAC" -t elastalert $CONFIGOPTIONS $BACKENDOPTIONS -o "$ELASTALERTRULES/$i" "$SIGMARULES/$i"
        ID=$(echo "$i" | cut -d"." -f1)
        csplit -z --prefix "misp_$ID" --suffix-format "_%04d.yml" "$ELASTALERTRULES/$i" "/^alert:/" "{*}" > /dev/null
        rm -f "$ELASTALERTRULES/$i"
done

echo "Done!"
```

**Comment by Jackson-Pollock** [comment]
Insane!! Thanks a lot for taking time to write this. This is working for me as expected. Will convert this to cron job as you did for [securityonion-misp](https://github.com/weslambert/securityonion-misp)

**Comment by weslambert** [comment]
No problem! Went ahead and created a GH repo (https://github.com/weslambert/misp2elastalert) for it in case it would be of use for anyone else.  Will update and make a little more streamlined as time allows.
